### PR TITLE
Alex-android merging to share entities

### PR DESCRIPTION
**Pull request by AlexLeonardos**

### CODE REVIEW

File: app/src/main/java/com/example/csc207courseproject/data_access/APIDataAccessObject.java [hunk 1]
```diff
@@ -273,4 +274,9 @@ public void setSeeding(List<Integer> seededEntrants) {
             throw new RuntimeException(event);
         }
     }
+
```
Comment (SahilBas2005) [inline app/src/main/java/com/example/csc207courseproject/data_access/APIDataAccessObject.java:277]: Add javadoc for this method

File: app/src/main/java/com/example/csc207courseproject/entities/Game.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package com.example.csc207courseproject.entities;
+
+
```
Comment (SahilBas2005) [inline app/src/main/java/com/example/csc207courseproject/entities/Game.java:3]: Add javadoc to all the methods in this class

File: app/src/main/java/com/example/csc207courseproject/entities/SetData.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
+package com.example.csc207courseproject.entities;
+
+
```
Comment (SahilBas2005) [inline app/src/main/java/com/example/csc207courseproject/entities/SetData.java:3]: Add javadoc to all the methods in this class

File: app/src/main/java/com/example/csc207courseproject/entities/Game.java (hunk 2 above)
Comment (SahilBas2005) [inline app/src/main/java/com/example/csc207courseproject/entities/Game.java:3]: Add javadoc to all the methods in this class

File: app/src/main/java/com/example/csc207courseproject/data_access/APIDataAccessObject.java (hunk 1 above)
Comment (SahilBas2005) [inline app/src/main/java/com/example/csc207courseproject/data_access/APIDataAccessObject.java:277]: Add javadoc to this method

File: app/src/main/java/com/example/csc207courseproject/entities/TournamentSets.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.example.csc207courseproject.entities;
+
```
Comment (SahilBas2005) [inline app/src/main/java/com/example/csc207courseproject/entities/TournamentSets.java:2]: I don't think this class is necessary as you can just pull ongoing sets into the state from the DAO

File: app/src/main/java/com/example/csc207courseproject/ui/report/ReportFragment.java [hunk 5]
```diff
@@ -1,35 +1,47 @@
 package com.example.csc207courseproject.ui.report;
```
Comment (SahilBas2005) [inline app/src/main/java/com/example/csc207courseproject/ui/report/ReportFragment.java:2]: Make sure you remove the ViewModelProvider from the framework and use our view model class

File: app/src/test/java/com/example/csc207courseproject/use_case/report_set/ReportSetInteractorTest.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+//
+//package use_case.report_set;
```
Comment (SahilBas2005) [inline app/src/test/java/com/example/csc207courseproject/use_case/report_set/ReportSetInteractorTest.java:2]: Is this meant to be completely commented out?

File: app/src/test/java/com/example/csc207courseproject/use_case/update_seeding/UpdateSeedingInteractorTest.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
+
+package com.example.csc207courseproject.use_case.update_seeding;
```
Comment (SahilBas2005) [inline app/src/test/java/com/example/csc207courseproject/use_case/update_seeding/UpdateSeedingInteractorTest.java:2]: Nice test code!!!! I love test coverage : - )

**Review by SahilBas2005 — APPROVED** [review]

File: app/src/main/java/com/example/csc207courseproject/data_access/APIDataAccessObject.java [hunk 8]
```diff
@@ -16,7 +17,7 @@
 import java.util.concurrent.CountDownLatch;
 
 public class APIDataAccessObject implements SelectPhaseDataAccessInterface, MainDataAccessInterface,
```
Comment (om0611) [inline app/src/main/java/com/example/csc207courseproject/data_access/APIDataAccessObject.java:19]: add Javadoc for this method

File: app/src/main/java/com/example/csc207courseproject/entities/Game.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    public String getPlayer2Character() {
+        return player2Character;
+    }
+
+    public boolean isReported() {
+        return winnerID != null;
+    }
+}
```
Comment (om0611) [inline app/src/main/java/com/example/csc207courseproject/entities/Game.java:37]: Add comments for each class method to make it more understandable to other group members.

File: app/src/main/java/com/example/csc207courseproject/entities/TournamentSets.java [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+public class TournamentSets {
+
+    private SetData[] ongoingSets;
+
+    private SetData[] upcomingSets;
+
+    public TournamentSets(SetData[] ongoingSets, SetData[] upcomingSets) {
```
Comment (om0611) [inline app/src/main/java/com/example/csc207courseproject/entities/TournamentSets.java:9]: Add javadoc comment for this method.

File: app/src/main/java/com/example/csc207courseproject/interface_adapter/report_game/ReportGameController.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+import com.example.csc207courseproject.interface_adapter.report_set.ReportSetState;
+import com.example.csc207courseproject.use_case.report_game.ReportGameInputBoundary;
+
+
+public class ReportGameController {
+    private final ReportGameInputBoundary reportGameUseCaseInteractor;
+    private final ReportSetState setState;
+
+    public ReportGameController(ReportGameInputBoundary reportGameUseCaseInteractor, ReportSetState setState) {
```
Comment (om0611) [inline app/src/main/java/com/example/csc207courseproject/interface_adapter/report_game/ReportGameController.java:11]: Add javadoc comment for this method.

File: app/src/main/java/com/example/csc207courseproject/interface_adapter/report_game/ReportGameController.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
+package com.example.csc207courseproject.interface_adapter.report_game;
+
+import com.example.csc207courseproject.interface_adapter.report_set.ReportSetState;
+import com.example.csc207courseproject.use_case.report_game.ReportGameInputBoundary;
+
+
+public class ReportGameController {
```
Comment (om0611) [inline app/src/main/java/com/example/csc207courseproject/interface_adapter/report_game/ReportGameController.java:7]: Add javadoc comment for this class.

File: app/src/main/java/com/example/csc207courseproject/interface_adapter/report_game/ReportGamePresenter.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+    private final ViewManagerModel viewManagerModel;
+
+    public ReportGamePresenter(ReportViewModel gameViewModel, ViewManagerModel viewManagerModel) {
+        this.gameViewModel = gameViewModel;
+        this.viewManagerModel = viewManagerModel;
+    }
+
+    @Override
+    public void prepareSuccessView(ReportGameOutputData outputData) {
+//        final SeedingState setState = gameViewModel.getState();
```
Comment (om0611) [inline app/src/main/java/com/example/csc207courseproject/interface_adapter/report_game/ReportGamePresenter.java:22]: Are these lines supposed to be commented out?

**Review by om0611 — APPROVED** [review]

**Pull request merged**